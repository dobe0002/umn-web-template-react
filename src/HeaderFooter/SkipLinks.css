.umnHeaderSkipLinksWrapperDiv {
  position: absolute;
  top: 0;
  left: -10000px;
  width: 1px;
  height: 1px;
  overflow: hidden;
  padding: 15px;
  background: #fff;
  border-radius: 0 0 5px 5px;
  box-shadow: 1px 2px 4px rgba(0, 0, 0, 0.5);
}

.umnHeaderSkipLinksWrapperDiv:focus-within {
  left: 0;
  width: auto;
  height: auto;
  overflow: auto;
}

.umnHeaderSkipLinksUl {
  margin: 0;
  padding: 0;
}

.umnHeaderSkipLinksLi {
  display: inline-block;
  margin-right: 15px;
}
