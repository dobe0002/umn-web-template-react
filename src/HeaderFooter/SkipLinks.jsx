import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Wrapper = styled.div`
  position: absolute;
  top: 0;
  left: -10000px;
  width: 1px;
  height: 1px;
  overflow: hidden;
  padding: 15px;
  background: #fff;
  border-radius: 0 0 5px 5px;
  box-shadow: 1px 2px 4px rgba(0, 0, 0, 0.5);
  &:focus-within {
    left: 0;
    width: auto;
    height: auto;
    overflow: auto;
  }
`;
const Ul = styled.ul`
  margin: 0;
  padding: 0;
`;
const Li = styled.li`
  display: inline-block;
  margin-right: 15px;
`;

const SkipLinks = props => {
  const { mainNavId, mainContentId } = props;
  if (props.mainNavId || props.mainContentId) {
    return (
      <Wrapper className='headerLinks'>
        <Ul>
          {mainNavId && (
            <Li>
              <a href={`#${mainNavId}`}>Main navigation</a>
            </Li>
          )}
          {mainContentId && (
            <Li>
              <a href={`#${mainContentId}`}>Main content</a>
            </Li>
          )}
        </Ul>
      </Wrapper>
    );
  }
  return '';
};

SkipLinks.propTypes = {
  mainNavId: PropTypes.string,
  mainContentId: PropTypes.string
};

export default SkipLinks;
