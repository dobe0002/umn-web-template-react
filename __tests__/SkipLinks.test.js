import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import SkipLinks from '../src/HeaderFooter/SkipLinks';
import Adapter from 'enzyme-adapter-react-16';
const { axe, toHaveNoViolations } = require('jest-axe');
expect.extend(toHaveNoViolations);
configure({ adapter: new Adapter() });

describe('Skip Links', () => {
  it('Accessibility check', async () => {
    const temp = (
      <div>
        <SkipLinks mainNavId='foo' mainContentId='bar' />
        <a id='foo' />
        <a id='bar' />
      </div>
    );
    const skipLinks = mount(temp);
    const results = await axe(skipLinks.getDOMNode());

    console.log('Axe violations', results.violations);

    expect(results).toHaveNoViolations();
  });
  it('Nothing is displayed when no props are passed', () => {
    const skipLinks = mount(<SkipLinks />);
    expect(skipLinks.html().trim()).toEqual('');
  });
  it('Only main nav skip links is it is only prop passed', () => {
    const skipLinks = shallow(<SkipLinks mainNavId='foo' />);
    // console.log(skipLinks.html());
    expect(skipLinks).toMatchSnapshot();
  });
  it('Only main content skip links is it is only prop passed', () => {
    const skipLinks = shallow(<SkipLinks mainContentId='bar' />);
    // console.log(skipLinks.html());
    expect(skipLinks).toMatchSnapshot();
  });
  it('Both skip links are displayed when passed', () => {
    const skipLinks = shallow(
      <SkipLinks mainNavId='foo' mainContentId='bar' />
    );
    // console.log(skipLinks.html());
    expect(skipLinks).toMatchSnapshot();
  });
});
