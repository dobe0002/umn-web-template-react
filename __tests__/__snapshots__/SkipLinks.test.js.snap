// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skip Links Both skip links are displayed when passed 1`] = `
<styled.div
  className="headerLinks"
>
  <styled.ul>
    <styled.li>
      <a
        href="#foo"
      >
        Main navigation
      </a>
    </styled.li>
    <styled.li>
      <a
        href="#bar"
      >
        Main content
      </a>
    </styled.li>
  </styled.ul>
</styled.div>
`;

exports[`Skip Links Only main content skip links is it is only prop passed 1`] = `
<styled.div
  className="headerLinks"
>
  <styled.ul>
    <styled.li>
      <a
        href="#bar"
      >
        Main content
      </a>
    </styled.li>
  </styled.ul>
</styled.div>
`;

exports[`Skip Links Only main nav skip links is it is only prop passed 1`] = `
<styled.div
  className="headerLinks"
>
  <styled.ul>
    <styled.li>
      <a
        href="#foo"
      >
        Main navigation
      </a>
    </styled.li>
  </styled.ul>
</styled.div>
`;
